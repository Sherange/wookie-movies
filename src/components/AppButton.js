import React from 'react';
import PropTypes from 'prop-types';
import {View, StyleSheet} from 'react-native';
import {Button} from 'react-native-elements';

const AppButton = props => {
  const {title} = props;
  return (
    <View style={styles.buttonContainer}>
      <Button
        title={title}
        type="outline"
        // buttonStyle={{width: '100%'}}
        containerStyle={{width: '100%'}}
      />
    </View>
  );
};

AppButton.propTypes = {
  title: PropTypes.string.isRequired,
};

const styles = StyleSheet.create({
  buttonContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    marginVertical: 8,
  },
});

export default AppButton;
