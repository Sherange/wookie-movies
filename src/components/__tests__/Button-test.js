import 'react-native';
import React from 'react';
import AppButton from '../AppButton';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  renderer.create(<AppButton title={'Search'} />);
});

test('AppButton snapshot', () => {
  const snapshot = renderer.create(<AppButton title={'Search'} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
