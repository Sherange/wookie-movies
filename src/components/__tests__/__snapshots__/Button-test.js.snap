// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppButton snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 8,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        Object {
          "width": "100%",
        },
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0.5,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2089dc",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Search
        </Text>
      </View>
    </View>
  </View>
</View>
`;
